engine/shaderc/src/shaderc_spirv.cpp

```diff
@@ -0,0 +1,177 @@
+        if (res_input)
+            return res_input;
+        const ShaderResource* res_output = FindShaderResource(context->m_Reflection.m_Outputs, name_hash);
+        if (res_output)
+            return res_output;
+        return 0;
+    }
+
+    static const ShaderResource* FindShaderResourceUniform(HShaderContext context, uint64_t name_hash)
+    {
+        const ShaderResource* res_ubo = FindShaderResource(context->m_Reflection.m_UniformBuffers, name_hash);
+        if (res_ubo)
+            return res_ubo;
+        const ShaderResource* res_ssbo = FindShaderResource(context->m_Reflection.m_StorageBuffers, name_hash);
+        if (res_ssbo)
+            return res_ssbo;
+        const ShaderResource* res_texture = FindShaderResource(context->m_Reflection.m_Textures, name_hash);
+        if (res_texture)
+            return res_texture;
+        return 0;
+    }
+
+    void SetResourceLocationSPIRV(HShaderContext context, ShaderCompilerSPIRV* compiler, uint64_t name_hash, uint8_t location)
+    {
+        const ShaderResource* res = FindShaderResourceInputOutput(context, name_hash);
+        if (res)
+        {
+            AddRemapEntry(compiler->m_RemapResources, name_hash, res->m_Id, res->m_Location, location, ShaderCompilerSPIRV::RemapResourceEntry::TYPE_LOCATION);
+        }
+    }
+
+    void SetResourceBindingSPIRV(HShaderContext context, ShaderCompilerSPIRV* compiler, uint64_t name_hash, uint8_t binding)
+    {
+        const ShaderResource* res = FindShaderResourceUniform(context, name_hash);
+        if (res)
+        {
+            AddRemapEntry(compiler->m_RemapResources, name_hash, res->m_Id, res->m_Binding, binding, ShaderCompilerSPIRV::RemapResourceEntry::TYPE_BINDING);
+        }
+    }
+
+    void SetResourceSetSPIRV(HShaderContext context, ShaderCompilerSPIRV* compiler, uint64_t name_hash, uint8_t set)
+    {
+        const ShaderResource* res = FindShaderResourceUniform(context, name_hash);
+        if (res)
+        {
+            AddRemapEntry(compiler->m_RemapResources, name_hash, res->m_Id, res->m_Set, set, ShaderCompilerSPIRV::RemapResourceEntry::TYPE_SET);
+        }
+    }
+
+    static ShaderCompilerSPIRV::RemapResourceEntry* FindEntryById(dmArray<ShaderCompilerSPIRV::RemapResourceEntry>& entries, uint32_t id, ShaderCompilerSPIRV::RemapResourceEntry::Type type)
+    {
+        for (int i = 0; i < entries.Size(); ++i)
+        {
+            if (entries[i].m_Type == type && entries[i].m_Id == id)
+            {
+                return &entries[i];
+            }
+        }
+        return 0;
+    }
+
+    ShaderCompileResult* CompileSPIRV(HShaderContext context, ShaderCompilerSPIRV* compiler, const ShaderCompilerOptions& options)
+    {
+        uint32_t* source = (uint32_t*) compiler->m_SPIRVCode;
+        uint32_t num_words_total = compiler->m_SPIRVCodeSize / sizeof(uint32_t);
+        uint32_t spirv_header_size_in_words = 5; // 20 bytes
```

Comment:
How is "word" defined here (or rather by spir-v)?

I'm used to the Win32 WORD (16 bits), and DWORD (32 bits).
But I'm not sure there is a C standard for it.

Is there a another name for it?

Reply:
Good call, it's taken from here:
https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Word

I'll add to the code